src/main/java/com/example/wdmsystem/employee/system/EmployeeService.java

```diff
@@ -1,46 +1,67 @@
 package com.example.wdmsystem.employee.system;
 
-import com.example.wdmsystem.exception.InvalidInputException;
+import com.example.wdmsystem.employee.system.authentication.CustomUserDetails;
 import com.example.wdmsystem.exception.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Limit;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 import java.time.LocalDateTime;
-import java.util.Arrays;
 import java.util.List;
 
 @Service
 public class EmployeeService {
     private final IEmployeeRepository employeeRepository;
 
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
     public EmployeeService(IEmployeeRepository employeeRepository) {
         this.employeeRepository = employeeRepository;
     }
 
-    public Employee createEmployee(EmployeeDTO request) {
+    public Employee createEmployee(CreateEmployeeDTO request) {
+        CustomUserDetails currentUser = (CustomUserDetails) SecurityContextHolder.getContext()
+                .getAuthentication()
+                .getPrincipal();
 
+        String hashedPassword = passwordEncoder.encode(request.password());
         Employee employee = new Employee(
                 0,
-                0, // placeholder
+                currentUser.getMerchantId(),
                 request.firstName(),
                 request.lastName(),
                 request.employeeType(),
                 request.username(),
-                request.password(),
+                hashedPassword,
                 LocalDateTime.now()
         );
         employee.setUpdatedAt(employee.getCreatedAt());
 
         return employeeRepository.save(employee);
     }
```

Comment:
When we merge, consder taking these parts where we convert DTOs to models and vice versa and putting them into the new DTOMApper class that I made, for consistency sake. If applicable of course.

Reply:
I will try to implement it